llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -122,6 +122,8 @@ Type *VPTypeAnalysis::inferScalarTypeForRecipe(const VPWidenRecipe *R) {
   case Instruction::FNeg:
   case Instruction::Freeze:
     return inferScalarType(R->getOperand(0));
+  case Instruction::ExtractValue:
+    return R->getUnderlyingInstr()->getType();
```

Comment:
This may return a vector type from `inferScalarTypeForRecipe` now?

Reply:
I think this should return a struct of scalars (so a "narrow" type rather than a scalar).

Reply:
Hmm, would it work to infer the scalar type of the operand and then extract the type for the index?

Reply:
Yes that works, and now that the extract indices are in the vplan, that does not require looking at the underlying value. 

P.s. I was wrong before, this should always return a scalar for `extractvalue`, as this patch only allow widening structs of scalars.